src/app/@ansyn/plugins/openlayers/maps/open-layers-map/openlayers-map/openlayers-map.ts

```diff
@@ -127,33 +175,64 @@ export class OpenLayersMap extends BaseImageryMap<OLMap> {
 		});
 	}
 
-	public resetView(layer: any, position: ICaseMapPosition, extent?: CaseMapExtent): Observable<boolean> {
-		this.isValidPosition = false;
-		const rotation = this._mapObject.getView() && this.mapObject.getView().getRotation();
+	public resetView(layer: any, position: ICaseMapPosition, extent?: CaseMapExtent, mapId?: string, useDoubleBuffer?: boolean): Observable<boolean> {
+		useDoubleBuffer = useDoubleBuffer && !layer.get(ImageryLayerProperties.FROM_CACHE);
+		if (useDoubleBuffer) {
+			this._backgroundMapObject = new OLMap(this._backgroundMapParams);
+		} else if (this._backgroundMapObject) {
+			this._backgroundMapObject.setTarget(null);
+			this._backgroundMapObject = null;
+		}
+		const rotation: number = this._mapObject.getView() && this.mapObject.getView().getRotation();
 		const view = this.createView(layer);
-		this.setMainLayer(layer);
-		this._mapObject.setView(view);
-
 		// set default values to prevent map Assertion error's
 		view.setCenter([0, 0]);
 		view.setRotation(rotation ? rotation : 0);
 		view.setResolution(1);
+		this.mapId = mapId;
+		if (useDoubleBuffer) {
+			this.setMainLayerToBackgroundMap(layer);
+			this._backgroundMapObject.setView(view);
+			this.savedParams = {
```

Comment:
add interface with the types (in this class)

Reply:
done